### ISSUE TITLE
Bug: incompatible states in trace handler

### ISSUE THREAD

**Issue by mikevoronov** [fluencelabs/aquavm#204]
AquaVM fails with
```
{
    "retCode": 20000,
    "errorMessage": "values in call results are not equal: Scalar(Object({})) != Scalar(String(\"\"))",
    "nextPeerPks": [],
    "callRequests": []
}
```

on certain input.

**Comment by mikevoronov** [comment]
The crux of this bug is a possibility to use streams in `match`/`mismatch` that was left for conviniece despite the non-determinism. The `match` instructions in PoC is used in order to implement the timeout pattern:
```
...
  (match $status.$.[0]! "done"
     ...
  )
...
```
and leads to this undeterministic behaviour.

In the future streams in `match`/`mismatch` will be prohibited in favor of streams canonicalizations and gates.

**Comment by mikevoronov** [comment]
Fixed by #297.
